mammal/examples/scrna_cell_type/pl_data_module.py

```diff
@@ -0,0 +1,263 @@
+    else:
+        stratify = None
+
+    train_ids, valid_ids = train_test_split(
+        h5ad_file.obs,
+        test_size=test_size,
+        random_state=random_state,
+        stratify=stratify,
+    )
+    train_adata = h5ad_file[train_ids.index]
+    validata_adata = h5ad_file[valid_ids.index]
+    return train_adata, validata_adata
+
+
+def load_datasets(
+    data_path: str | Path = "data", stratify_by=None
+) -> dict[str, DatasetDefault]:
+
+    data_path = Path(data_path)
+    if not data_path.is_absolute():
+        data_path = Path(__file__).parent / data_path
+    anndata_object = anndata.read_h5ad(data_path)
+    anndata_dict = {}
+    anndata_dict["all_data"] = anndata_object
+    anndata_dict["all_train"], anndata_dict["test"] = anndata_train_test_split(
+        anndata_dict["all_data"],
+        test_size=0.1,
+        stratify_by=stratify_by,
+    )
+    anndata_dict["train"], anndata_dict["valid"] = anndata_train_test_split(
+        anndata_dict["all_train"],
+        test_size=0.1 / (1.0 - 0.1),
+        random_state=2024,
+        stratify_by=stratify_by,
+    )
+
+    ds_dict = {}
+    for set_name in ["train", "valid", "test"]:
+        input_anndata = anndata_dict[set_name]
+        size = input_anndata.shape[0]
+        print(f"{set_name} set size is {size}")
+
+        dynamic_pipeline = PipelineDefault(
+            "cell_type",
+            [
+                (OpReadAnnData(input_anndata), {"prefix": "scrna"}),
+            ],
+        )
+
+        ds = DatasetDefault(sample_ids=size, dynamic_pipeline=dynamic_pipeline)
+        ds.create()
+        ds_dict[set_name] = ds
+
+    return ds_dict
+
+
+def load_cell_type_mapping(
+    mapping_key="cell_type",
+    mapping_value="cell_type_ontology_term_id",
+    cell_type_mapping="cell_type_mapping.csv",
+):
+    """
+    Load metadata_extra_mapping.csv from the given dataset metadata folder,
+    and return the values of a requested key and value columns as a dictionary.
+
+    This is used to convert the names from the ones in the input anndata to the
+    ones that are known to the tokenizer.
+    """
+    cell_type_mapping_file_path = Path(__file__).parent / cell_type_mapping
+
+    if not os.path.exists(cell_type_mapping_file_path):
+        raise FileNotFoundError(str(cell_type_mapping_file_path) + "is not found")
+    else:
+        mapping_df = pd.read_csv(cell_type_mapping_file_path, index_col=False)
+        cell_type_mapping = dict(
+            zip(
+                mapping_df[mapping_key],
+                mapping_df[mapping_value],
+            )
+        )
+        return cell_type_mapping
+
+
+def preprocess_ann_data(
+    anndata_object: anndata.AnnData,
+    min_genes: int = 200,
+    num_bins: int = 10,
+    cell_type: str = "cell_type",
+):
+    """run preprocessing steps on anndata object
+    assumes that the anndata object has a standard structure with counts per cell X gene, and cell type annotations in obs["cell_type"].
```

Comment:
```suggestion
    assumes that the anndata object has a standard structure with counts per cell X gene, and cell type annotations in obs[cell_type].
```
We want to refer to the variable value here